kong/db/strategies/cassandra/init.lua

```diff
@@ -280,6 +286,9 @@ local function serialize_arg(field, arg)
   elseif field.type == "record" then
     serialized_arg = cassandra.text(cjson.encode(arg))
 
+  elseif field.type == "foreign" then
+    serialized_arg = cassandra.uuid(arg[field.schema.primary_key[1]])
```

Comment:
This is still assuming that the key will always be of type `uuid`. I suggest resolving the field and recursing this function instead (untested)

```lua
local fk_pk = field.schema.primary_key[1]
local fk_field = field.schema.fields[pk]
serialized_arg = serialize_arg(fk_field, arg[fk_pk])
```

Reply:
changed accordingly